cores/arduino/Adafruit_TinyUSB_Core/Adafruit_USBD_Device.cpp

```diff
@@ -254,7 +254,7 @@ static int strcpy_uni16(const char *s, uint16_t *buf, int bufsize) {
   int i = 0;
   int buflen = 0;
 
-  while (i < bufsize) {
+  while (s[i] > 0 && i < bufsize) {
```

Comment:
thanks for the PR, though should the extra condition is `s[i] != 0` if we are trying to find the null terminator ?

Reply:
Oh, I am used to ```char``` being ```unsigned```. You are right, checking  ```!=0``` should be safer and work in any case.

Reply:
@kaysievers yeah, could you push the update so that we could merge and fix this :).

Reply:
Rebased and pushed with the fix. Tested it and dumped the descriptor, looks like the string lengths are fine:
```
String Descriptor Table
--------------------------------
Index  LANGID  String
0x00   0x0000  0x0409 
Hex dump: 
0x04 0x03 0x09 0x04 

0x01   0x0409  "V2"
Hex dump: 
0x06 0x03 0x56 0x00 0x32 0x00 

0x02   0x0409  "pad"
Hex dump: 
0x08 0x03 0x70 0x00 0x61 0x00 0x64 0x00 

0x03   0x0409  "226B0C055333395336202020FF093B13"
Hex dump: 
0x42 0x03 0x32 0x00 0x32 0x00 0x36 0x00 0x42 0x00 
0x30 0x00 0x43 0x00 0x30 0x00 0x35 0x00 0x35 0x00 
0x33 0x00 0x33 0x00 0x33 0x00 0x33 0x00 0x39 0x00 
0x35 0x00 0x33 0x00 0x33 0x00 0x36 0x00 0x32 0x00 
0x30 0x00 0x32 0x00 0x30 0x00 0x32 0x00 0x30 0x00 
0x46 0x00 0x46 0x00 0x30 0x00 0x39 0x00 0x33 0x00 
0x42 0x00 0x31 0x00 0x33 0x00 
```